src/reconcile/rec_visibility.c

```diff
@@ -248,6 +248,79 @@ __timestamp_no_ts_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw)
     return (false);
 }
 
+/*
+ * __rec_delete_hs_upd --
+ *     Delete the value left in the history store
+ */
+static int
+__rec_delete_hs_upd(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ROW *rip, uint64_t recno,
+  WT_UPDATE *delete_upd, WT_UPDATE *delete_tombstone)
+{
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_ITEM hs_recno_key;
+    uint8_t *p, hs_recno_key_buf[WT_INTPACK64_MAXSIZE];
+    bool hs_read_committed_set;
+
+#ifdef HAVE_DIAGNOSTIC
+    WT_TIME_WINDOW *hs_tw;
+#endif
+
+    /* Open a history store cursor if we don't yet have one. */
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+    hs_read_committed_set = F_ISSET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    if (r->page->type == WT_PAGE_ROW_LEAF) {
+        WT_ERR(__wt_scr_alloc(session, 0, &key));
+        WT_ERR(__wt_row_leaf_key(session, r->page, rip, key, true));
+        r->hs_cursor->set_key(r->hs_cursor, 4, S2BT(session)->id, key, WT_TS_MAX, UINT64_MAX);
+    } else {
+        p = hs_recno_key_buf;
+        WT_ERR(__wt_vpack_uint(&p, 0, recno));
+        hs_recno_key.data = hs_recno_key_buf;
+        hs_recno_key.size = WT_PTRDIFF(p, hs_recno_key_buf);
+        r->hs_cursor->set_key(
+          r->hs_cursor, 4, S2BT(session)->id, &hs_recno_key, WT_TS_MAX, UINT64_MAX);
+    }
+    WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_before(session, r->hs_cursor), true);
+    /* It's possible the value in the history store is already obsolete. */
+    if (ret == WT_NOTFOUND) {
+        ret = 0;
+        goto done;
+    }
+
+#ifdef HAVE_DIAGNOSTIC
+    __wt_hs_upd_time_window(r->hs_cursor, &hs_tw);
+    WT_ASSERT(session, hs_tw->start_txn == WT_TXN_NONE || hs_tw->start_txn == delete_upd->txnid);
+    WT_ASSERT(session, hs_tw->start_ts == WT_TS_NONE || hs_tw->start_ts == delete_upd->start_ts);
+    WT_ASSERT(session,
+      hs_tw->durable_start_ts == WT_TS_NONE || hs_tw->durable_start_ts == delete_upd->durable_ts);
+    if (delete_tombstone != NULL) {
+        WT_ASSERT(
```

Comment:
if the stop timestamp is NONE, this record shouldn't be visible right? Do we really need to check for this condition?

Reply:
HS record is always visible regardless of tombstone. I'm not sure what you mean.

Reply:
Fixed in 3e2bd757bdf59bfccc77b1370ced624cce320626